internal/adapters/dataproviders/rootstock/pegout_contract.go

```diff
@@ -73,6 +74,34 @@ func (pegoutContract *pegoutContractImpl) IsPegOutQuoteCompleted(quoteHash strin
 	return result, nil
 }
 
+func (pegoutContract *pegoutContractImpl) ValidatePegout(quoteHash string, btcTx []byte) error {
+	var quoteHashBytes [32]byte
+	// Set From address to the LP provider address so msg.sender validation passes
+	opts := &bind.CallOpts{
+		From: pegoutContract.signer.Address(),
+	}
+	hashBytesSlice, err := hex.DecodeString(quoteHash)
+	if err != nil {
+		return err
+	} else if len(hashBytesSlice) != 32 {
+		return errors.New("quote hash must be 32 bytes long")
+	}
+	copy(quoteHashBytes[:], hashBytesSlice)
+	_, err = pegoutContract.contract.ValidatePegout(opts, quoteHashBytes, btcTx)
+	if err != nil {
+		// Try to parse the revert reason to distinguish contract validation errors from network/RPC errors
+		parsedRevert, parseErr := ParseRevertReason(pegoutContract.abis.PegOut, err)
+		if parseErr != nil && parsedRevert == nil {
```

Comment:
I think this is enough and less complex
```suggestion
		if parseErr != nil {
```